src/createTokenBridge.ts

```diff
@@ -0,0 +1,213 @@
+  const txHash = await parentChainPublicClient.sendRawTransaction({
+    serializedTransaction: await rollupOwner.signTransaction(txRequest),
+  });
+
+  // get the transaction receipt after waiting for the transaction to complete
+  const txReceipt = createTokenBridgePrepareTransactionReceipt(
+    await parentChainPublicClient.waitForTransactionReceipt({ hash: txHash }),
+  );
+  console.log(
+    `Deployed in ${getBlockExplorerUrl(parentChainPublicClient.chain)}/tx/${
+      txReceipt.transactionHash
+    }`,
+  );
+
+  // wait for retryables to execute
+  console.log(`Waiting for retryable tickets to execute on the Orbit chain...`);
+  const orbitChainRetryableReceipts = await txReceipt.waitForRetryables({
+    orbitPublicClient: orbitChainPublicClient,
+  });
+  console.log(`Retryables executed`);
+  console.log(
+    `Transaction hash for first retryable is ${orbitChainRetryableReceipts[0].transactionHash}`,
+  );
+  console.log(
+    `Transaction hash for second retryable is ${orbitChainRetryableReceipts[1].transactionHash}`,
+  );
+
+  // fetching the TokenBridge contracts
+  const tokenBridgeContracts = await txReceipt.getTokenBridgeContracts({
+    parentChainPublicClient,
+  });
+
+  // Non custom fee token
+  if (!nativeTokenAddress) {
+    // set weth gateway
+    const setWethGatewayTxRequest = await createTokenBridgePrepareSetWethGatewayTransactionRequest({
+      rollup: rollupAddress,
+      parentChainPublicClient,
+      orbitChainPublicClient,
+      account: rollupOwner.address,
+      ...createTokenBridgePrepareRegisterWethGatewayTransactionRequestParamsOverride,
+    });
+
+    // sign and send the transaction
+    const setWethGatewayTxHash = await parentChainPublicClient.sendRawTransaction({
+      serializedTransaction: await rollupOwner.signTransaction(setWethGatewayTxRequest),
+    });
+
+    // get the transaction receipt after waiting for the transaction to complete
+    const setWethGatewayTxReceipt = createTokenBridgePrepareSetWethGatewayTransactionReceipt(
+      await parentChainPublicClient.waitForTransactionReceipt({ hash: setWethGatewayTxHash }),
+    );
+
+    console.log(
+      `Weth gateway set in ${getBlockExplorerUrl(parentChainPublicClient.chain)}/tx/${
+        setWethGatewayTxReceipt.transactionHash
+      }`,
+    );
+
+    // Wait for retryables to execute
+    const orbitChainSetWethGatewayRetryableReceipt =
+      await setWethGatewayTxReceipt.waitForRetryables({
+        orbitPublicClient: orbitChainPublicClient,
+      });
+    console.log(`Retryables executed`);
+    console.log(
+      `Transaction hash for retryable is ${orbitChainSetWethGatewayRetryableReceipt[0].transactionHash}`,
+    );
+    if (orbitChainSetWethGatewayRetryableReceipt[0].status !== 'success') {
+      throw new Error(
+        `Retryable status is not success: ${orbitChainSetWethGatewayRetryableReceipt[0].status}. Aborting...`,
+      );
+    }
+  }
+
+  return tokenBridgeContracts;
```

Comment:
Let's also return other data, here similar to how we do in [createRollup](https://github.com/OffchainLabs/arbitrum-orbit-sdk/blob/main/src/createRollup.ts#L198). I think we can add:

- transaction
- transactionReceipt
- retryables (just tx hashes is fine)
- tokenBridgeContracts
- setWethGateway { transaction, transactionReceipt, retryables }

Reply:
We should make this a type and export it